c2corg_ui/static/js/viewdetails.js

```diff
@@ -304,37 +428,38 @@ app.ViewDetailsController.prototype.initPhotoswipe_ = function() {
 
 
 /**
- * @private
- */
-app.ViewDetailsController.prototype.initSlickGallery_ = function() {
-  $('.photos').slick({});
-};
-
-
-/**
  * @export
  */
 app.ViewDetailsController.prototype.getComments = function() {
   var topic_id = this.documentService.document['topic_id'];
   if (topic_id === null) {
     return;
   }
-
-  // create DiscourseEmbed script tag. From a discourse tutorial.
-  // https://meta.discourse.org/t/embedding-discourse-comments-via-javascript/31963
-  var s = document.createElement('script');
-  /**
-   * @export
-   * @type {appx.DiscourseEmbedded}
-   */
-  window.DiscourseEmbed = {
-    'discourseUrl': this.discourseUrl_,
-    'topicId': topic_id
-  };
-  s.src = this.discourseUrl_ + 'javascripts/embed.js';
-  document.getElementsByTagName('body')[0].appendChild(s);
+  var document = this.documentService.document;
+  var lang = document.lang;
+  this.api_.readCommentsForum(topic_id,lang).then(function(response) {
+    this.handleCommentsForum(response);
+  }.bind(this), function() { // Error msg is shown in the api service
+  }.bind(this));
 };
 
+/**
+ * Handles forum processing for comments
+ * @param response
+ * @public
+ */
+app.ViewDetailsController.prototype.handleCommentsForum = function(response) {
+  var data = response['data'];
```

Comment:
You almost only use ``data['post_stream']`` => that would make more sense to have a shortcut for it, not for ``response['data']``.

```
if ('post_stream' in response['data']) {
  var posts = response['data']['post_stream']['posts'];
  for (var i = 0, n = posts.length; i < n; i++) {
    if (posts[i]['name'] == 'system') {
    ...
  } 
}
```